staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -542,11 +542,12 @@ func (config *inClusterClientConfig) RawConfig() (clientcmdapi.Config, error) {
 }
 
 func (config *inClusterClientConfig) ClientConfig() (*restclient.Config, error) {
-	if config.inClusterConfigProvider == nil {
-		config.inClusterConfigProvider = restclient.InClusterConfig
```

Comment:
this removes a side-effect, right? So formally it is a breaking change, although to the better.

@liggitt wdyt?

Reply:
this looks fine. the type is private, the member field is only used from this method (outside of unit tests), and the default impl (`restclient.InClusterConfig`) is stateless